controllers/utils/hash.go

```diff
@@ -34,6 +35,19 @@ func CreateUniqueSecretName(managedCluster, storageClusterNamespace, storageClus
 	return CreateUniqueName(managedCluster, storageClusterNamespace, storageClusterName)[0:39]
 }
 
-func CreateUniqueReplicationId(fsids []string) string {
-	return CreateUniqueName(fsids...)[0:39]
+func CreateUniqueReplicationId(clusterFSIDs map[string]string) (string, error) {
+	var fsids []string
+	for _, v := range clusterFSIDs {
+		if v != "" {
+			fsids = append(fsids, v)
+		}
+	}
+
+	if len(fsids) < 2 {
+		return "", fmt.Errorf("replicationID can not be generated due to missing cluster FSID")
```

Comment:
Add which cluster ID we are actually missing. It should be present as the key in `clusterFSIDs`

Reply:
There could be multiple and I didn't want to do a bunch of string manipulation. Since this is a very corner case, can we keep that as a follow up item?

Reply:
or it can done in the place where we are fetching the clusterFSIDs. It should be easier and will make more sense to log that